### ISSUE TITLE
Smarter Analysis Limits

### ISSUE THREAD

**Issue by KyleMiles** [Vector35/binaryninja-api#2714]
- [ ] Tag functions that trigger analysis limits with the reason they were canceled.
- [ ] Don't trigger analysis limits on User Functions
- [ ] Don't trigger analysis limits on functions created from Debug Sources
- [ ] Don't trigger analysis limits on "high confidence" functions - recursive decent, very high confidence linear sweep functions

Possibly add another analysis phase where we go back and analyze functions that triggered certain limits, eg. functions that exceeded the maximum analysis time.

**Comment by ccarpenter04** [comment]
I really like these ideas

**Comment by jduck** [comment]
I agree. Good idea to queue beefier analysis up for later. Maybe retry them with temporarily increased limits.
